d1985r3.md

```diff
@@ -376,148 +303,114 @@ limitations we saw before `if constexpr` was introduced: Function templates
 often had to rely on helper structs to do simple things.
 
 
-### Allowing unversal template parameters in code
+### Dependent Names
 
 To make this feature easier to use we also propose that universal template
```

Comment:
This is from my recent PR so maybe there was a mistake with "late" above. Lets handle that later.

Reply:
Yeah, your recent PR handled things differently from what I'm proposing. This PR merges/rewrites our two PRs to use the "dependent names as template arguments are UTPs" rule. This means that the subsection "Deferring kind checking to instantiation" is outdated because UTPs/dependent names are promoted to UTPs when used as template arguments. There kind *is* a UTP in the parsing context. The UTP is narrowed to a type or value in other contexts but is the root kind here.

Reply:
It seems most of the changes that were in this PR are gone, so your comment here above is not valid, I think. As for the title none of them is actually good. I rather wish we had defined that we will use UTP throughout to avoid the recurring long "universal template parameter" everywhere. That would bring my title into reasonable length at least.